### ISSUE TITLE
Move weights of fitness function to a parameter of `train.py`

### ISSUE THREAD

**Issue by CCRcmcpe** [ultralytics/yolov5#7176]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar feature requests.

### Description

All "weights" below is "weights of fitness function" for short.

Recently I made recall value more important halfway through a training process on Kaggle.
I must write a script to modify metrics.py, and open the latest model, change `best_fitness` of it to 0, then save it. Otherwise "best model" is stuck to previous best fitness value, and no more "best model" will be saved, because the new fitness function is always less than previous one.  
  
Not quite convenient in my opinion. Weights may vary according to the user's wish. They not saved to wandb either, making them difficult to track.

### Use case

Whenever the user want to make recall value, precision value, or mAP@0.5,0.95 more important in a training process or a hyperparameters evolve process, the user can use `--fitness-weights` parameter or something, also the weights are saved to wandb.

### Additional

To solve the previous `best_fitness` problem we can implement the solution above in code.

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@CCRcmcpe thanks for the feedback! 

I'd recommend starting a new training if you modify your training settings. Changing settings in the middle of training may have unintended consequences.

Also beware that Recall is an arbitrary metric and should not be weighed heavily in the fitness function, if at all. This is because Recall is up to you, it's not a property of the model, but a property that you set yourself by varying the confidence threshold. You can create 100% recall for any model on any dataset by simply lowering --conf to zero.

**Comment by CCRcmcpe** [comment]
@glenn-jocher  Thank you for your hint. Weight recall more is to save the model which acheived the best recall while keeping the confidence threshold unchanged, afaik, so the model will predict less false negatives on test set.

Change the weights halfway is not the main goal, but a mistake in my training process, change weights in train.py parameters instead of the code is. Then, is moving fitness weights to train.py parameters, and keep it tracked on wandb a good idea?

**Comment by glenn-jocher** [comment]
@CCRcmcpe absolutely, I appreciate the clarification! We aim to keep the codebase flexible and user-friendly, so parameterization of fitness weights in train.py, as well as tracking them on Weights & Biases, makes for a solid feature addition. It's beneficial for users to have a straightforward way to customize fitness weights and track their effects. Thanks for the insightful suggestion!
